dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -82,6 +82,18 @@ def individual_steps(steps)
   wait_until {@browser.title rescue nil}
 end
 
+When /^I open a new tab$/ do
+  @browser.execute_script('window.open();')
+end
+
+When /^I close the current tab$/ do
+  @browser.close
+end
+
+When /^I switch to tab index (\d+)$/ do |tab_index|
+  @browser.switch_to.window(@browser.window_handles[tab_index.to_i])
```

Comment:
I wonder if this will have trouble in some of our UI test browser configurations.  Have you run this against Firefox/IE/Safari?